src/Service/MailService.php

```diff
@@ -94,15 +95,38 @@ public function send(): ResultInterface
         }
 
         //save copy of sent message to folders
-        if ($this->mailOptions->getSaveSentMessageFolder()) {
+        if ($this->mailOptions->getTransport() == 'Laminas\Mail\Transport\Smtp' &&
+            $this->mailOptions->getSaveSentMessageFolder()) {
             foreach ($this->mailOptions->getSaveSentMessageFolder() as $folder) {
-                $this->storage->appendMessage($this->message->toString(), $folder);
+                $this->storage = $this->createStorage();
+                if ($this->storage) {
+                    $this->storage->appendMessage($this->message->toString(), $folder);
+                }
             }
         }
 
         return $result;
     }
 
+    /**
```

Comment:
Would rearrange this method like this, because we should:
* use as many early returns as possible
* check config variables with `empty` instead of checking against empty string (this avoids eventual bugs if one edits the config file and uses `null`/`false` instead of empty strings)
* avoid saving the `Imap` instance in a variable if on the next line we return it

```
/**
 * @return Imap|null
 */
protected function createStorage(): ?Imap
{
    $host = $this->mailOptions->getSmtpOptions()->getHost();
    if (empty($host)) {
        return null;
    }
    $connectionConfig = $this->mailOptions->getSmtpOptions()->getConnectionConfig();
    if (empty($connectionConfig['username']) || empty($connectionConfig['password'])) {
        return null;
    }

    return new Imap([
        'host'     => $host,
        'user'     => $connectionConfig['username'],
        'password' => $connectionConfig['password'],
    ]);
}
```

Please test after applying these changes.

Reply:
yes, it still works with these updates